public/pages/CreateDetector/components/ConfigureFieldMapping/components/RequiredFieldMapping/FieldMappingsTable.tsx

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { Component } from 'react';
+import { RouteComponentProps } from 'react-router-dom';
+import {
+  EuiBasicTableColumn,
+  EuiEmptyPrompt,
+  EuiIcon,
+  EuiInMemoryTable,
+  EuiText,
+} from '@elastic/eui';
+import { DEFAULT_EMPTY_DATA } from '../../../../../../utils/constants';
+import { STATUS_ICON_PROPS } from '../../utils/constants';
+import SIEMFieldNameSelector from './SIEMFieldName';
+import { FieldMappingsTableItem } from '../../../../types/interfaces';
+
+interface FieldMappingsTableProps extends RouteComponentProps {
+  loading: boolean;
+  isMappingRequired: boolean;
+  indexFields: string[];
+  aliasNames: string[];
+}
+
+interface FieldMappingsTableState {
+  remainingUnmappedAlias: Set<string>;
+}
+
+export default class FieldMappingsTable extends Component<
+  FieldMappingsTableProps,
+  FieldMappingsTableState
+> {
+  constructor(props: FieldMappingsTableProps) {
+    super(props);
+    this.state = {
+      remainingUnmappedAlias: new Set(props.aliasNames),
+    };
+  }
+
+  static getDerivedStateFromProps(props: FieldMappingsTableProps): FieldMappingsTableState {
+    return {
+      remainingUnmappedAlias: new Set(props.aliasNames),
+    };
+  }
+
+  componentDidMount = async () => {};
```

Comment:
Is this necessary?

Reply:
Not right now, I guess I refactored the code inside and left it empty, let me remove it

Reply:
sounds good!

Reply:
removed for now